packages/core/core/src/requests/EntryRequest.js

```diff
@@ -102,42 +102,74 @@ class EntryResolver {
 
     if (stat.isDirectory()) {
       let pkg = await this.readPackage(entry);
-      if (pkg && pkg.source != null) {
+
+      if (pkg) {
+        let {filePath} = pkg;
         let entries = [];
-        let files = [];
-
-        let pkgSources = Array.isArray(pkg.source) ? pkg.source : [pkg.source];
-        for (let pkgSource of pkgSources) {
-          if (typeof pkgSource === 'string') {
-            let source = path.join(path.dirname(pkg.filePath), pkgSource);
-            try {
-              stat = await this.options.inputFS.stat(source);
-            } catch (err) {
-              throw new ThrowableDiagnostic({
-                diagnostic: {
-                  message: `${pkgSource} in ${path.relative(
-                    this.options.inputFS.cwd(),
-                    pkg.filePath,
-                  )}#source does not exist`,
-                  filePath: source,
-                },
-              });
+        let files = [{filePath}];
+
+        let targetsWithSources = 0;
+        if (pkg.targets) {
+          for (let targetName in pkg.targets) {
+            let target = pkg.targets[targetName];
+            if (target.source != null) {
+              targetsWithSources++;
+              let targetSources = Array.isArray(target.source)
+                ? target.source
+                : [target.source];
+              for (let source of targetSources) {
+                let filePath = path.join(entry, source);
+                entries.push({
```

Comment:
Should we do the same validation here as for the top level source field? Eg that the path exists and is a file

Reply:
I refactored the exist/file checks into a shared function and now apply them to both. Also added a new EntryRequest.test.js file and added unit tests for both of those checks on package#source and package#targets.*.source values.
<img width="1289" alt="Screen Shot 2021-02-23 at 1 43 23 PM" src="https://user-images.githubusercontent.com/783032/108912908-1c204b00-75de-11eb-903a-2e53066abb9b.png">